### DISCUSSION TITLE
CSS Calendar Source

### DISCUSSION THREAD

**Discussion by mcharltonau** [totaldebug/atomic-calendar-revive#1540] [Feature Requests]
Add the calendar source to the event-title class.  e.g. <div class="event-title family" style=".......
Will enable card mod to change the title colors based on calendar source.

**Comment by filip-heens** [comment]
would also be nice for the appointments... so that we can look in a single glance whose appointment it is ;-)

**Comment by davislp** [comment]
> Add the calendar source to the event-title class. e.g. <div class="event-title family" style="....... Will enable card mod to change the title colors based on calendar source.

It's not ideal, but you can set different icons on different calendars and then use card mod to make the different icons different colors

```
card_mod:
  style: |
    ha-icon[icon="mdi:circle"] {
      color: aqua !important;
    }
    ha-icon[icon="mdi:triangle"] {
      color: teal !important;
    }
```

**Reply by dml105** [reply]
You can also set the icon field to not be an actual icon like mdi:circle. I set my icons to "Dad" "Mom" "Son" and "Daughter" to separate our individual calendars, then set the colors. 

Then you can use the "has" selector to select the entire event and mod all you like, even deleting the icon.

```
.cal-eventContainer {
      --dadColor: #00cc00;
      --dadColorT: #00cc0044;
      --momColor: #cc0000;
      --momColorT: #cc000044;
      --sonColor: #0000ff;
      --sonColorT: #0000ff44;
      --daughterColor:#ffcc00;
      --daughterColorT:#ffcc0044;
      --eventSize: 16px;
      --time-size: 10px;
      --locationSize: 10px;
      .event-icon {display: none};
    }
    .cal-card {padding: 6px !important}
    .single-event-container {gap:0px !important; padding-bottom:6px !important;}
    .daywrap {border-top:none !important; ;};
    .event-right .ha-icon {display: none;}
    .event-right-top { 
        display: contents !important;
        .event-location {
            text-align: right;
            display: inline-block;
            vertical-align: top;
            user-select: text;
            overflow-wrap: anywhere;
            font-size: var(--locationSize);
    }
    }
    .event-right:has(ha-icon[icon="dad"]) {
      font-size: var(--eventSize);
      padding: 8px;
      border: var(--dadColor) 1px solid;
      border-radius: 8px; 
      background-color: var(--dadColorT); 
      border-left-width: 12px;
      .event-icon {display: none};
    }

    .event-right:has(ha-icon[icon="mom"]) {
      font-size: var(--eventSize);
      padding: 8px;
      border: var(--momColor) 1px solid;
      border-radius: 8px; 
      background-color: var(--momColorT); 
      border-left-width: 12px;
      .event-icon {display: none};
    }

    .event-right:has(ha-icon[icon="son"]) {
      font-size: var(--eventSize);
      padding: 8px;
      border: var(--sonColor) 1px solid;
      border-radius: 8px; 
      background-color: var(--sonColorT); 
      border-left-width: 12px;
      .event-icon {display: none};
    }

    .event-right:has(ha-icon[icon="daughter"]) {
      font-size: var(--eventSize);
      padding: 8px;
      border: var(--daughterColor) 1px solid;
      border-radius: 8px; 
      background-color: var(--daughterColorT); 
      border-left-width: 12px;
      .event-icon {display: none};
    }
```

<img width="334" alt="Screenshot 2025-01-11 at 1 04 51 AM" src="https://github.com/user-attachments/assets/648e8dfe-20c8-4166-b414-ac2ce67a0cf0" />

**Comment by marksie1988** [comment]
Implemented on the `beta` branch in c5206a1. Every `.event-title` now carries a `.cal-<slug>` class derived from the calendar entity_id — `calendar.family_main` becomes `cal-family-main` (lowercased, non-alphanumeric collapsed to dashes). Target it with card_mod:

```yaml
card_mod:
  style: |
    .event-title.cal-family {
      border-left: 3px solid var(--accent-color);
      padding-left: 6px;
    }
    .event-title.cal-holidays-in-united-kingdom {
      color: #4caf50;
    }
```

See the new "Semantic classes on event titles" section in the styling docs for the full list (`event-title-fullday`, `running`, `recurring`, `Event/Calendar/Planner`, `cal-<slug>`). Closing as implemented.
